src/v/cloud_storage/cache_service.cc

```diff
@@ -378,11 +391,25 @@ ss::future<> cache::trim() {
     if (size_to_delete > 0 || objects_to_delete > 0) {
         vlog(
           cst_log.info,
-          "trim: regular trim did not free enough space, executing exhaustive "
+          "trim: fast trim did not free enough space, executing exhaustive "
           "trim to free {}/{}...",
           size_to_delete,
           objects_to_delete);
-        co_await trim_exhaustive(size_to_delete, objects_to_delete);
+        auto exhaustive_result = co_await trim_exhaustive(
+          size_to_delete, objects_to_delete);
+        size_to_delete -= std::min(
+          exhaustive_result.deleted_size, size_to_delete);
+        objects_to_delete -= std::min(
+          exhaustive_result.deleted_count, objects_to_delete);
+        if (size_to_delete || objects_to_delete) {
```

Comment:
nit: seems like this could be just `exhaustive_result.deleted_size < size_to_delete` etc